packages/frontend-core/src/fetch/DataFetch.ts

```diff
@@ -130,26 +211,26 @@ export default class DataFetch {
    * Fetches a fresh set of data from the server, resetting pagination
    */
   async getInitialData() {
-    const { datasource, filter, paginate } = this.options
+    const { filter, paginate } = this.options
 
     // Fetch datasource definition and extract sort properties if configured
-    const definition = await this.getDefinition(datasource)
+    const definition = await this.getDefinition()
 
     // Determine feature flags
-    const features = this.determineFeatureFlags(definition)
+    const features = await this.determineFeatureFlags()
     this.features = {
       supportsSearch: !!features?.supportsSearch,
       supportsSort: !!features?.supportsSort,
       supportsPagination: paginate && !!features?.supportsPagination,
     }
 
-    // Fetch and enrich schema
-    let schema = this.getSchema(datasource, definition)
```

Comment:
Is it safe to remove the enrichment and just check the base definition?

Reply:
I think this is actually not needed, but added it back just in case. It could be required for views v1